utils/customize_node.py

```diff
@@ -1149,6 +1156,9 @@ def execute(args):
         update_status(args, 'Untar golden image')
         args.new_fs_dir = untar(args.build_dir + '/untar/', args.golden_tar)
 
+        if args.hostname != 'golden': #Keep kernel when building a golden image
```

Comment:
Another place that needs the new boolean attribute.   We don't need three ways to identify the phase.

Reply:
I think we don't even need "phase_msg" variable in extract_bootfiles because of this line, actually.
Idea is, if it is a golden image build - skip extraction of boot files, right? So this line does that.
"phase_msg" would serve no purpose anymore.

And the way it was before:

>
>extract_bootfiles(args, 'golden')
>...
>install_packages
>...
>extract_bootfiles(args, 'add-on')

this is a bug. It will always extract bootfiles from golden image build, because the last extract_bootfiles tells it to.
We need to know if it is a golden build or node build by looking at 'args.hostname'. That way bootfiles will stay in golden, but will be extracted during node build.

Reply:
Never-mind. I converted phase_msg to bool instead and it should be alright like that.